### PR TITLE
fix(desktop): make the unreachable-deck pane render, and date what it shows (#1143)

### LINKED ISSUE #1143: Desktop: the unreachable-deck no-terminal state cannot render, because such a deck reports no agents
## What

PRD #1105 decided that an agent pane whose deck has no live link **"replaces its terminal with a sentence rather than closing the pane"** (`prds/1105-desktop-agent-pane-overlay.md`). Only the second half of that is true in production today: the pane is not closed, but the sentence never renders.

## Why it cannot render

The no-terminal state is part of the pane, and the pane needs an **agent record** to render anything at all — its header, role, status, prompt, tool and five panel tabs are all properties of that record.

A deck with no live link reports **no agents**, so there is no record to render:

- `desktop/src-tauri/src/dto.rs:1708` — `disconnected_snapshot` carries `agents: Vec::new()`.
- `desktop/src-tauri/src/daemon_bridge.rs:1237` — `snapshot_with`'s non-connected early return does the same.
- Those are the only two empty agent lists under `desktop/src-tauri/src/`, and both are on the non-connected path.
- `mapDesktopSnapshot` (`desktop/src/lib/bridge.ts`) builds `agents` from `dto.agents` alone and carries none over from the previous snapshot, and `pruneFleet` keeps a configured-but-unreachable deck in the fleet with an empty list.

So on a deck that stops answering, the agent lookup fails and the pane renders nothing, rather than rendering the state that was built for exactly this case.

## What currently holds instead

PR #1126 guarantees the **view survives** an unreachable deck (`paneAgentRetired` is gated on `paneDeckAttachable`), so the pane returns intact when the deck answers again. That is asserted by `keeps the view when the pane's deck stops answering, so the pane returns when it answers again` in `desktop/src/AgentPaneNoTerminal.test.tsx`.

## Note on the existing test

`replaces an unreachable deck's terminal with an explicit state naming that deck` passes because its harness keeps `agents` populated on a disconnected deck — which the crate never does. The assertion is valid **given a record**; it just does not prove the state is reachable. Worth revisiting with this fix rather than before it.

## What fixing it implies

Holding the last known agent record for an open pane. That is a product call rather than a mechanical fix: how stale a header may be, and what on screen says so. Deliberately not taken in #1126.

Found during the review of PR #1126.

### PR DESCRIPTION
**Pull request by vfarcic**
## The defect

PRD #1105 decided that an agent pane whose deck has no live link **"replaces its terminal with a sentence rather than closing the pane"**. Only the second half was true in production: the pane was not closed, but the sentence never rendered.

The pane is built entirely out of the agent **record** — header, role, status, prompt, tool, five panel tabs — and a deck with no live link reports **no agents**. `disconnected_snapshot` (`desktop/src-tauri/src/dto.rs:1716`) and `snapshot_with`'s non-connected early return (`desktop/src-tauri/src/daemon_bridge.rs:1195`) both carry `agents: Vec::new()`; those are the only two empty agent lists under `desktop/src-tauri/src/` and both are on the non-connected path, and `mapDesktopSnapshot` builds `agents` from `dto.agents` alone. So `paneAgent` was `undefined`, `DeckShell`'s render condition failed, and the state built for exactly this case never rendered.

## The product call, both halves

Fixing it means holding the last known record for an open pane, which the issue is explicit is a product decision rather than a mechanical fix.

### How stale may a header be — **unbounded in time, bounded by the pane**

The record lives exactly as long as the pane the user opened, and only for that pane's `(deckId, agentId)`. There is **no expiry timer**, and that is a decision:

- An expiry would make the pane vanish at a moment the app picked and the user cannot see. That is the failure `paneAgentRetired` already refuses one condition away — it declines to close on a **silence**, because a silence is not an answer — and it would be worse here, because it would fire after the user had been looking at the pane.
- A thirty-second outage and a thirty-minute one are the same state of knowledge: the app knows nothing in both. Any threshold would be an arbitrary point at which "unknown" is redefined as "gone".
- So the pane **says how old the record is** rather than deciding on the reader's behalf. A timestamp is information; a timeout is a guess.

The hold is dropped on close, on a change of pane identity, and the moment a live record returns.

### What on screen says so — **three affordances, and the first is load-bearing**

Showing a stale header with nothing saying it is stale asserts a liveness this app has no evidence for, which the issue judges worse than showing nothing.

1. **The status stops asserting.** It is the field that lies hardest: a bare `running` beside a live teal dot is a claim about *now*. Held, the chip reads `last seen: <status>` and both colour cues drop to neutral. `last seen: passed` rather than `last seen passed` — half the status values are not verbs. The heading, role and identity are left alone: they are what the agent **is**, not what it is doing, and they do not decay.
2. **One machine-readable seam.** `data-agent-record="held" | "live"` on the tile's `<article>`, a **sibling** of `data-status` rather than a value inside it — the held record still *has* a status, one that was true earlier, so folding `held` into the status vocabulary would make everything that keys on a status have to read it. Same shape `data-terminal-state` already set: one attribute plus one `role="status"` sentence.
3. **The sentence dates the report.** The no-terminal notice gains *"The rest of this pane is what `<deck>` last reported 4m ago, and nothing in it is being updated"*, in the app's own `displayActivity` vocabulary rather than a second set of buckets, with the exact instant on the notice's `title`. The relative label needs no timer of its own: an unreachable deck's watcher takes `spawn_deck_watcher`'s no-link arm every iteration, emitting a snapshot *before* sleeping `WATCH_RETRY_DELAY` (1s), `emit_snapshot` is a bare `app.emit` with no dedupe, and `adoptFleet` rebuilds the fleet on each one. The `title` needs none of that argument — an absolute instant cannot decay.

### The deck-origin pane is deliberately not held, and the reason is measured

Raised by the PR reviewer as a P1 and answered rather than deferred. The factual half is correct: the held record reaches `OverviewAgentPane` and nothing else, so a deck-origin pane whose selected deck disconnects still renders nothing.

A promoted pane marks everything that is not one of its own ancestors `inert` — `useInertBackground`, whose doc names *"a connection banner"* as exactly the background content it re-marks on every commit. So a held deck-origin pane would explain that the deck is not answering **while putting the one control that fixes it behind an inert barrier**. Measured on this PR's own fixture rather than reasoned about: with a deck-origin pane promoted and a banner showing, the banner carries `inert` and its `Reconnect` button inherits it.

The two screens are not symmetrical, which is why one answer does not fit both. The overview goes on listing every other deck's agents normally, so the missing pane is the only surface that can say what happened, and `Esc` returns to a screen that is still useful. The deck screen says it itself, in a `role="alert"` banner carrying that deck's own failure message and the `Reconnect` / `Start deck` remedy, and renders **no tiles at all**, so there is nothing a reader could misread as live. What that path keeps is PR #1126's guarantee: the view is not closed, so the pane is back the moment the deck answers.

`leaves a deck-origin pane's screen to explain itself, with its remedy reachable` pins both halves as a connected → disconnected → connected round trip, and reads `inert` off the banner so the scope cannot be reversed silently.

### The limit, named rather than hidden

A pane whose deck has **never** answered renders nothing. That is not a hold that failed — it is a record that does not exist anywhere in the app, and rendering one would mean inventing it. It is also not a path a user reaches by pressing anything: the overview lists no agents for a deck that is not answering, so the Open control is not on the screen. The same reasoning covers a deck that has left the **observed set** entirely (`paneDeck === undefined`), which keeps today's behaviour: the state needs that deck's own account of why it is not answering, and a deck the app is no longer watching gives none — synthesising one would be the app inventing a failure it did not observe, the distinction `pruneFleet` already *"refuses to cross on its own"*. A new test asserts this limit and the recovery from it.

## The misleading test, fixed

`replaces an unreachable deck's terminal with an explicit state naming that deck` was green while the behaviour it named was unreachable, because its harness kept `agents` populated on a disconnected deck and the crate never does. A harness that models a failure more kindly than production certifies the thing it was built to catch.

`disconnected` now empties the list unconditionally, which makes the file load-bearing for the hold rather than green whatever it does. Consequences:

- `replaces an unreachable deck's terminal…` now runs against the real path and asserts all three affordances.
- `attaches the terminal and clears the notice when the pane's own deck answers again` became a healthy → away → healthy **round trip**. It used to start cold on a disconnected deck, which the corrected harness makes unrenderable and which was never a path a user could reach; the round trip also proves the half a cold start could not — that the hold is **surrendered** when a live record arrives.
- `gives a non-selected but reachable deck's agent a live terminal instead` gained `data-agent-record="live"`, so the hedging cannot leak to a live record.

**PR #1126's guarantee is kept and strengthened.** `keeps the view when the pane's deck stops answering, so the pane returns when it answers again` still passes, still on `paneAgentRetired` being gated on `paneDeckAttachable`, and now also asserts the pane is **rendered** rather than merely un-navigated-away-from. Its second `harness` argument went, because `disconnected` empties the list on its own — which is what that test always claimed to be modelling. `paneAgentRetired` and the shown declaration both still read the **live** `paneAgent`, never the held one, so a connected deck retiring an agent still closes the view and nothing is ever declared shown for a record nothing can attach to.

## Rule 18 — weighed, and the answer is client-side

The record wanted is one this app already received and discarded, and the daemon that owns it is **by definition the unreachable one**. No additive optional field, no capability-gated verb and no wire-shape change can be answered by a process that is not answering, and no other daemon in the fleet knows anything about that one's agents. There is no rung on the graded path to reach for: this is a **retention** decision, not a data-availability one.

It is also correctly in the webview rather than in the crate, which is the one place a component-boundary flinch could have hidden. Scoping the hold to the open pane requires knowing which pane is open, and that is frontend state the crate does not have; holding it crate-side would mean putting last-known agents back into `disconnected_snapshot`, where every consumer of `snapshot.agents` picks them up and the overview starts listing agents a deck cannot vouch for — contradicting PRD #1105 directly.

## Rule 12 — no

TypeScript, TSX, CSS and Markdown only. `src/daemon_protocol.rs` is untouched, no frame, field or verb is added or changed, no Tauri command is added, and nothing reaches the daemon, the protocol, orchestration or hooks. **No `PROTOCOL_VERSION` bump, no `.breaking.md` fragment, no `CONTRACT_BREAKS` entry, and the cross-version manual test is not owed.**

## Verification

| gate | result |
| --- | --- |
| `cargo fmt --check` | clean |
| `cargo clippy --workspace --all-targets --features e2e,e2e-live -- -D warnings` | clean |
| `cargo test-fast` | **3299 passed**, 0 skipped |
| `desktop` `tsc --noEmit` | clean |
| `desktop` `pnpm test` (vitest) | **617 passed**, 26 files (604 before) |
| `desktop` `pnpm test:browser` (playwright) | **132 passed** (unchanged) |

**Tests covering what was touched, by name and count** — the surface is the desktop webview, which the Rust `tests/CATALOG.md` tiers do not cover, so these are the Vitest and Playwright suites:

- `desktop/src/AgentPaneNoTerminal.test.tsx` — **10**, the file this change is about (+2 new: `renders no pane for a deck that has never answered, and brings one back when it does`, and `leaves a deck-origin pane's screen to explain itself, with its remedy reachable`).
- `desktop/src/hooks/useHeldAgentRecord.test.ts` — **6**, new.
- `desktop/src/lib/terminalInput.test.ts` — **22** (+5 for `unreachableDeckTerminalState`).
- `desktop/src/AgentPaneOverlay.test.tsx` — **9**; `desktop/src/AgentPaneDeckIdentity.test.tsx` — **9**; `desktop/src/CrossDeckTerminalState.test.tsx` — **5**; `desktop/src/App.test.tsx` — **71**; `desktop/src/lib/displayText.test.ts` — **35**. All other consumers of `AgentTile`/`DeckShell`/`displayActivity`.
- Playwright: the full 132, which includes `e2e/agent-pane-overlay.spec.ts` and `e2e/overview-row-open.spec.ts`.

**Mutation-checked, four times.** Each applied alone, run, reverted, and the tree verified back to green. Counts are measured, not expected:

1. **Identity fence dropped** (return the held record without comparing the key) → **2 red**, exactly the two fence tests.
2. **The hold never surrendered** (return it on renders that carry a live record too) → **5 red** across both files, including the reachable-deck control, whose header starts hedging under a live terminal.
3. **`recordFreshness` pinned to `"live"`** at the pane's call site → **3 red**.
4. **The harness's populated agent list restored on a disconnected deck** — the original blind spot, reproduced deliberately → **4 red**, which is *more* than the one test it was expected to break: with the crate's real empty list modelled, the affordance test, the round trip and #1126's guarantee test all depend on the hold too. The mutation that certified the shipped defect no longer passes anywhere.

Changelog fragment: `changelog.d/1143.bugfix.md`. The decision is written into `prds/1105-desktop-agent-pane-overlay.md` — the Work Log entry of 2026-09-19, plus the two places in the body that recorded the gap as open.

Closes #1143

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: desktop/src/App.tsx [hunk 1]
```diff
@@ -365,7 +398,7 @@ export function DeckShell({ runtime, workflowPlatformIssue, initialView = { kind
           one live `TerminalViewport` for the agent, which is the property M3
           actually requires.
         */}
-        {agentView && paneDeck && paneAgent && <OverviewAgentPane runtime={runtime} view={agentView} deck={paneDeck} agent={paneAgent} attached={paneDeckAttachable} onClose={closeAgent} />}
+        {agentView && paneDeck && paneAgentShown && <OverviewAgentPane runtime={runtime} view={agentView} deck={paneDeck} agent={paneAgentShown} held={heldPaneAgent} attached={paneDeckAttachable} onClose={closeAgent} />}
```
[the remark this answers is not in the document]
Comment (vfarcic) [reply desktop/src/App.tsx:414]: The factual half is correct and I have not changed it: the held record reaches `OverviewAgentPane` and nothing else, so a deck-origin pane whose selected deck disconnects still renders nothing. That is deliberate, and `8af10ae7` records the reason and pins it with a test rather than leaving it implicit — thank you for catching that it was undocumented.

**Why the deck path is not held, measured rather than argued.** A promoted pane marks everything that is not one of its own ancestors `inert` — `useInertBackground`, whose doc comment names *"a connection banner"* as exactly the background content it re-marks on every commit. So a held deck-origin pane would explain that the deck is not answering **while putting the one control that fixes it behind an inert barrier**. I measured it on this PR's own fixture rather than reasoning about it: with a deck-origin pane promoted and a banner showing, the banner carries `inert` and its `Reconnect` button inherits it down the flat tree.

**The two screens are not symmetrical, which is why one answer does not fit both.**

- The **overview** goes on listing every other deck's agents normally. Nothing on that screen says the pane's deck stopped answering, so the pane is the only surface that can — and `Esc` returns to a screen that is still useful.
- The **deck** screen says it itself, in a `role="alert"` banner carrying that deck's own failure message and the `Reconnect` / `Start deck` remedy, and it renders **no tiles at all**, so there is nothing a reader could misread as live. Covering that with a modal that inerts the remedy would be a net loss.

**What the deck path keeps is PR #1126's guarantee**, which is the part that matters for a blank screen: the view is not closed, so the pane is back intact the moment the deck answers.

`leaves a deck-origin pane's screen to explain itself, with its remedy reachable` (`desktop/src/AgentPaneNoTerminal.test.tsx`) is the round trip you asked for, asserted from the other side: connected → disconnected → connected, checking that no pane renders, the banner is present and **not** `inert`, its `Reconnect` is reachable, it carries the deck's own failure sentence, and the pane returns on reconnect. Reading `inert` off the banner is what stops the scope being reversed silently — extend the hold to the deck path and that test goes red with the reason attached. The decision is also written into `prds/1105-desktop-agent-pane-overlay.md` and into the `useHeldAgentRecord` call site in `App.tsx`.

Resolving on that basis. If you think the deck screen should nonetheless carry the pane, the thing to settle first is the inert conflict above — that is a change to `DeckSurface`'s promote-in-place model (PRD #1105 M3), not a prop.
